Core/clim-core/dialog-views.lisp

```diff
@@ -70,6 +70,37 @@
                                          :default default
                                          :default-type default-type))
 
+(defun make-gadget-pane-from-view (view stream query-identifier &rest initargs)
+  (let* ((class-view (type-of view))
+         (class-gadget (intern (subseq (symbol-name class-view) 0 (position #\- (symbol-name class-view) :from-end t)) 'climi))
+         (slots (c2mop:class-slots (find-class class-view))))
+    (loop for slot in slots when (slot-boundp view (c2mop:slot-definition-name slot)) do
+         (push (slot-value view (c2mop:slot-definition-name slot)) initargs)
+         (push (car (c2mop:slot-definition-initargs slot)) initargs))
```

Comment:
I would like a different solution for this. How about the following: a new generic function `gadget-for-view` return two values: a gadget class and a plist of initargs and values. A method specialized on e.g. `gadget-view' can then do something similar to this code as a default.

To get rid of the `intern` and symbol name mangling at runtime: the macro that generates the view classes could put the name of the corresponding gadget class into a class-allocated slot of the view class.

Reply:
I try to follow your suggestion, I hope that now it is better.